[8.0] add module purchase_origin_address

yvaucher opened the pull request:
In this module, we add a settable origin address on the purchase order that we copy on created picking in.

yvaucher committed:
add module purchase_origin_address

yvaucher committed:
split tests in travis

yvaucher commented:
something wrong in tests. Maybe some incompatibility with other modules -> WIP

yvaucher commented:
Last tests confirms an hidden incompatibility

pedrobaeza commented on .travis.yml:
```diff
@@ -8,6 +8,8 @@ env:
   - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE="framework_agreement"
   - VERSION="8.0" ODOO_REPO="odoo/odoo" EXCLUDE="framework_agreement"
   - VERSION="8.0" ODOO_REPO="OCA/OCB" EXCLUDE="framework_agreement"
+  - VERSION="8.0" ODOO_REPO="odoo/odoo" INCLUDE="purchase_origin_address"
```
This is not needed. Module is tested on the lines with EXCLUDE option, because it tests all modules except the one in the option. INCLUDE changes behaviour, because it only tests modules put on the argument.

yvaucher replied:
This is what I wanted to do. A test with only purchase_origin_address.

This WIP I wanted to check if on travis I had the same correct result as locally for the tests.

Now I have to figure out which is the guilt module. Unless I lazily adds purchase_origin_address tested in stand alone. But the incompatibility seams to ignore my additions. As partner is not forwarded

pedrobaeza replied:
Sorry, I didn't see WIP tag. I have made comments anyway later on code to help you :wink:

pedrobaeza commented on purchase_origin_address/README.rst:
```diff
@@ -0,0 +1,11 @@
+=======================
+
+This modules adds an origin address on the purchase order that will be
+transfered on picking in.
```
s/transfered/transferred

yvaucher replied:
Thanks, good catch

pedrobaeza commented on purchase_origin_address/model/purchase_order.py:
```diff
@@ -0,0 +1,46 @@
+    def onchange_partner_id(self, partner_id):
+        res = super(PurchaseOrder, self).onchange_partner_id(partner_id)
+        if not partner_id:
+            res['value']['origin_address_id'] = False
```
You can put directly:
`res['value']['origin_address_id'] = partner_id`

pedrobaeza commented on purchase_origin_address/model/stock_picking.py:
```diff
@@ -0,0 +1,29 @@
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    origin_address_id = fields.Many2one(
```
What if you make this field related and you don't have to assign it? Or do you want to change it later?

gurneyalex replied:
That's the idea yes : a given supplier may have different addresses from which the shipment may come from and so we need a m2o and not a related field.

lepistone commented on purchase_origin_address/tests/test_origin_address.py:
```diff
@@ -0,0 +1,119 @@
+
+        po2.signal_workflow('purchase_confirm')
+
+        self.assertEquals(po2.picking_ids.origin_address_id.id,
```
assertFalse

yvaucher replied:
Indeed

lepistone commented:
Hang on, I am improving one of the tests.

lepistone committed:
refactor the test of delivery address

Now setup is done in setUp, and the test methods contain only action +
assertion.

lepistone commented:
I refactored the tests.
 :+1: Thanks!

yvaucher commented:
@lepistone thanks

lepistone committed:
add myself to the contributors

lepistone committed:
remove checks: it is ignored on v8

gurneyalex commented:
:+1:
